src/compiler/builder.ts

```diff
@@ -0,0 +1,523 @@
+/// <reference path="program.ts" />
+
+namespace ts {
+    export interface EmitOutput {
+        outputFiles: OutputFile[];
+        emitSkipped: boolean;
+    }
+
+    export interface EmitOutputDetailed extends EmitOutput {
+        diagnostics: Diagnostic[];
+        sourceMaps: SourceMapData[];
+        emittedSourceFiles: SourceFile[];
+    }
+
+    export interface OutputFile {
+        name: string;
+        writeByteOrderMark: boolean;
+        text: string;
+    }
+
+    /* @internal */
+    export interface Builder {
+        /**
+         * Call this to feed new program
+         */
+        updateProgram(newProgram: Program): void;
+        getFilesAffectedBy(program: Program, path: Path): string[];
+        emitFile(program: Program, path: Path): EmitOutput;
+
+        /** Emit the changed files and clear the cache of the changed files */
+        emitChangedFiles(program: Program): EmitOutputDetailed[];
+        /** When called gets the semantic diagnostics for the program. It also caches the diagnostics and manage them */
+        getSemanticDiagnostics(program: Program, cancellationToken?: CancellationToken): Diagnostic[];
+
+        /** Called to reset the status of the builder */
+        clear(): void;
+    }
+
+    interface EmitHandler {

```

Comment:
this should be intrnal too

Reply:
consider spiting this into two `namespace ts {` declarations

Reply:
This isnt public interface anyways ?